pkg/drivers/sqlite/sqlite.go

```diff
@@ -105,3 +118,76 @@ func setup(db *sql.DB) error {
 
 	return nil
 }
+
+// alterTableIndices drops the given old table indices from the existing
+// kine table (if it exists, and if the indices exists), and then creates
+// the given new ones on the kine table.
+func alterTableIndices(d *generic.Generic) error {
+	if d.LockWrites {
+		d.Lock()
+		defer d.Unlock()
+	}
+
+	// Drop the old indices
+	for _, stmt := range dropIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create the new indices
+	for _, stmt := range createIndices {
+		_, err := d.DB.Exec(stmt)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// checkMigrate performs migration from an old key value table to the kine
+// table only if the old key value table exists and migration has not been
+// done already.
+func checkMigrate(ctx context.Context, d *generic.Generic) error {
+	row := d.DB.QueryRowContext(ctx, userVersionSQL)
+	if row == nil {
+		return fmt.Errorf("internal error: cannot find user_version pragma")
+	}
+
+	var userVersion int
+	if err := row.Scan(&userVersion); err != nil {
+		return err
+	}
+	// No need for migration
+	if userVersion == 1 {
+		return nil
+	}
+
+	// Check if the key_value table exists
+	row = d.DB.QueryRowContext(ctx, tableListSQL)
+	var tableCount int
+	if err := row.Scan(&tableCount); err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
An aggregate cannot return an empty set (= this check will never be true).

Reply:
Cheers - removed this check.